vdyp-forward/src/main/java/ca/bc/gov/nrs/vdyp/forward/VdypForwardApplication.java

```diff
@@ -0,0 +1,3207 @@
+
+import static ca.bc.gov.nrs.vdyp.forward.VdypPass.Pass1;
+import static ca.bc.gov.nrs.vdyp.forward.VdypPass.Pass2;
+import static ca.bc.gov.nrs.vdyp.forward.VdypPass.Pass3;
+import static ca.bc.gov.nrs.vdyp.forward.VdypPass.Pass4;
+import static ca.bc.gov.nrs.vdyp.forward.VdypPass.Pass5;
+
+import java.io.BufferedReader;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ca.bc.gov.nrs.vdyp.application.VdypApplication;
+import ca.bc.gov.nrs.vdyp.application.VdypApplicationIdentifier;
+
+public class VdypForwardApplication extends VdypApplication {
+
+	private static final Logger log = LoggerFactory.getLogger(VdypForwardApplication.class);
+
+	public static final int CONFIG_LOAD_ERROR = 1; // TODO check what Fortran VDYP Forward would exit with.
+	public static final int PROCESSING_ERROR = 2; // TODO check what Fortran VDYP Forward would exit with.
+
+	public static final String DEFAULT_VDYP_CONTROL_FILE_NAME = "vdyp.ctr";
+
+	private static Set<VdypPass> vdypPassSet = new HashSet<>(
+			Arrays.asList(new VdypPass[] { Pass1, Pass2, Pass3, Pass4, Pass5 })
+	);
+
+	public static void main(final String... args) {
+
+		var app = new VdypForwardApplication();
+
+		app.logVersionInformation();
+
+		List<String> controlFileNames = null;
+
+		try {
+			if (args.length == 0) {
+				System.out.print("Enter name of VDYP control file (or RETURN for vdyp.ctr) or *name for both): ");
+
+				controlFileNames = new ArrayList<>();
+				try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
+					String userResponse = br.readLine();
+					if (userResponse.length() == 0) {
+						controlFileNames.add(DEFAULT_VDYP_CONTROL_FILE_NAME);
+					} else if (userResponse.startsWith("*")) {
+						controlFileNames.add(DEFAULT_VDYP_CONTROL_FILE_NAME);
+
+						userResponse = userResponse.substring(1);
+						controlFileNames.addAll(Arrays.asList(userResponse.split("[[:space:]]+")));
+					}
+				}
+			} else {
+				controlFileNames = Arrays.asList(args);
+			}
+		} catch (Exception ex) {
+			log.error("Error during initialization", ex);
+			System.exit(CONFIG_LOAD_ERROR);
+		}
+
+		try {
+			VdypForwardProcessor processor = new VdypForwardProcessor(app, vdypPassSet, controlFileNames);
+			processor.process();
+		} catch (Exception ex) {
+			log.error("Error during processing", ex);
+			System.exit(PROCESSING_ERROR);
+		}
+	}
+
+	private void logVersionInformation() {
+		log.info(RESOURCE_SHORT_VERSION + " " + RESOURCE_VERSION_DATE);
+		log.info(RESOURCE_BINARY_NAME + " Ver:" + RESOURCE_SHORT_VERSION + " " + RESOURCE_VERSION_DATE);
+		log.info("VDYP7 Support Ver: " + AVERSION);
+	}
+
+	@Override
+	public VdypApplicationIdentifier getId() {
+		return VdypApplicationIdentifier.VDYPForward;
+	}
+
+//	public static final float PI_40K = 0.78539816E-04f;
```

Comment:
This class is incomplete. I left the original FipStart code in because I didn't know how much of it could be used here. It will be used if possible and otherwise removed.